File: DataGridview ComboBox usage/[C#][VB]-DataGridview ComboBox usage/C# and VB/Demo1_C/Extensions/DataExtensions.cs
using System.Data;

namespace Demo1_C.Extensions
{
    public static class DataExtensions
    {
        public static T LastValue<T>(this DataTable dt, string ColumnName)
        {
            return dt.Rows[dt.Rows.Count - 1].Field<T>(dt.Columns[ColumnName]);
        }
    }
}

File: DataGridview ComboBox usage/[C#][VB]-DataGridview ComboBox usage/C# and VB/Demo1/Extensions/DataExtensions.vb
Public Module DataExtensions
    <System.Diagnostics.DebuggerStepThrough()> _
    <System.Runtime.CompilerServices.Extension()> _
    Public Function LastValue(Of T)(ByVal dt As DataTable, ByVal ColumnName As String) As T
        Return dt.Rows.Item(dt.Rows.Count - 1).Field(Of T)(dt.Columns(ColumnName))
    End Function
End Module
